scout/ci.md

```diff
@@ -140,10 +141,230 @@ Add the following to the YAML file:
+
+It then downloads curl and the Docker CLI and logs into the Gitlab CI registry and the Docker registry using environment variables defined in your repository's settings.
+
+Add the following to the YAML file:
+
+{% raw %}
+```yaml
+script:
+  - |
+    if [[ "$CI_COMMIT_BRANCH" == "$CI_DEFAULT_BRANCH" ]]; then
+      tag=""
+      echo "Running on default branch '$CI_DEFAULT_BRANCH': tag = 'latest'"
+    else
+      tag=":$CI_COMMIT_REF_SLUG"
+      echo "Running on branch '$CI_COMMIT_BRANCH': tag = $tag"
+    fi
+  - docker build --pull -t "$CI_REGISTRY_IMAGE${tag}" .
+  - |
+    if [[ "$CI_COMMIT_BRANCH" == "$CI_DEFAULT_BRANCH" ]]; then
+      # Get a CVE report for the built image and fail the pipeline when critical or high CVEs are detected
+      docker scout cves "$CI_REGISTRY_IMAGE${tag}" --exit-code --only-severity critical,high    
+    else
+      # Compare image from branch with latest image from the default branch and fail if new critical or high CVEs are detected
+      docker scout compare "$CI_REGISTRY_IMAGE${tag}" --to "$CI_REGISTRY_IMAGE:latest" --exit-code --only-severity critical,high --ignore-unchanged
+    fi
+
+  - docker push "$CI_REGISTRY_IMAGE${tag}"
+```
+{% endraw %}
+
+This creates the flow mentioned previously. If the commit was to the default branch, Docker Scout generates a CVE report. If the commit was to a different branch, Docker Scout compares the new version to the current published version. It only shows critical or high-severity vulnerabilities and
+ignores vulnerabilities that haven't changed since the last analysis.
+
+Add the following to the YAML file:
+
+```yaml
+rules:
+  - if: $CI_COMMIT_BRANCH
+    exists:
+      - Dockerfile
+```
+
+These final lines ensure that the pipeline only runs if the commit contains a Dockerfile and if the commit was to the CI branch.
+
+## Microsoft Azure DevOps Pipelines
+
+The following examples runs in an Azure DevOps-connected repository containing a Docker image's definition and contents. Triggered by a commit to the main branch, the pipeline builds the image and uses Docker Scout to create a CVE report.
+
+Add the following to an _azure-pipelines.yml_ file.
+
+First, set up the rest of the workflow. Add the following to the YAML file:
+
+```yaml
+trigger:
+  - main
+
+resources:
+  - repo: self
+
+variables:
+  tag: "$(Build.BuildId)"
+  image: "vonwig/nodejs-service"
+```
+
+This sets up the workflow to use a particular container image for the application and tag each new image build with the build ID.
+
+Add the following to the YAML file:
+
+```yaml
+stages:
+  - stage: Build
+    displayName: Build image
+    jobs:
+      - job: Build
+        displayName: Build
+        pool:
+          vmImage: ubuntu-latest
+        steps:
+          - task: Docker@2
+            displayName: Build an image
+            inputs:
+              command: build
+              dockerfile: "$(Build.SourcesDirectory)/Dockerfile"
+              repository: $(image)
```

Comment:
What repository is this?

Reply:
my guess is it's the `variables.image` defined above (so here set to `vonwig/nodejs-service`)

Reply:
@eunomie That I get, but was more meaning what's the difference between the image defined in the Dockerfile and the repository mentioned?

Reply:
That's the same. `repository` is the parameter name for that step which takes an image name. The value comes from the variable as @eunomie said. I only extracted the repository into the `image` var to be able to reuse this later in the `docker scout cves $(image)` command.